network/hostport_test.go

```diff
@@ -510,3 +511,79 @@ func (*HostPortSuite) makeHostPorts() []network.HostPort {
 		)...,
 	)
 }
+
+const (
+	shortTimeout  = 250 * time.Millisecond
+	mediumTimeout = 4 * shortTimeout
+	longTimeout   = 5 * mediumTimeout
+	noTimeout     = 0
+)
+
+func (*HostPortSuite) TestFastestHostPortAllUnreachable(c *gc.C) {
+	// All of the endpoints below should either block indefinitely (without a
+	// timeout) or fail immediately.
+	unreachableHPs := network.NewHostPorts(49151, // IANA reserved port (unreachable)
+		"255.1.2.3",       // IPv4 route unreachable
+		"bad.example.com", // unresolvable
```

Comment:
On my system:

    $ dig bad.example.com +short
    92.242.132.15

Are you expecting this to not resolve?